src/it/scala/ie/zalando/fabric/gateway/ApiValidationSpec.scala

```diff
@@ -203,4 +203,19 @@ class ApiValidationSpec
       ingressii should have length 0
     }
   }
+
+  it should "Create routes for handling OPTIONS requests when CORS is enabled" in {
+    synchRequest(ValidSynchRequestWithCorsEnabled.payload) ~> Route.seal(createRoutesFromDerivations(ingressDerivationLogic)) ~> check {
+      val ingressii = responseAs[TestSynchResponse].ingressii
+      val corsRoutes = ingressii
+        .filter(_.name.contains("cors"))
+        .flatMap(_.route)
+      corsRoutes.size should be > 0
+      corsRoutes should contain theSameElementsAs List(
+        """Path("/api/resource") && Method("OPTIONS") && Header("X-Forwarded-Proto", "https") -> enableAccessLog(4, 5) -> status(204) -> flowId("reuse") -> corsOrigin("https://example.com", "https://example-other.com") -> appendResponseHeader("Access-Control-Allow-Methods", "POST, OPTIONS") -> appendResponseHeader("Access-Control-Allow-Headers", "X-Flow-Id, Content-Type") -> appendResponseHeader("Access-Control-Allow-Credentials", "true") -> <shunt>""",
```

Comment:
Does it make sense to be more fine grained in what we search for here? There's a lot of global predicates and filters here which would require the test to be updated if they changed.

Reply:
Shotgun surgery antipattern

Reply:
This is a custom route, it's not affected by filters added to other routes. The filters and predicates should only change if the cors feature itself is explicitly changed, at which point we want this test to break